lambdas/indexer/app.py

```diff
@@ -74,27 +74,28 @@ def hello():
     return {'Hello': 'World!'}
 
 
+@app.route('/delete', methods=['POST'])
 @app.route('/', methods=['POST'])
 def post_notification():
     """
-    Receive a notification event and either queue it for asynchronous indexing or process it synchronously.
+    Receive a notification event and queue it for indexing or deletion.
     """
     hmac.verify(current_request=app.current_request)
     notification = app.current_request.json_body
     log.info("Received notification %r", notification)
     validate_request_syntax(notification)
-    params = app.current_request.query_params
+    if app.current_request.context['path'] == '/':
+        return process_notification('add', notification)
+    elif app.current_request.context['path'].startswith('/delete'):
```

Comment:
This is technically correct, but confusing to a person reading this. I would do, 

```suggestion
    elif app.current_request.context['path'] in ('/delete', '/delete/')
```

And did you confirm that this is indeed needed? Does Chalice not strip the trailing slash before calling the handler?

We should also add `else: assert False` as we always do.

Reply:
@hannes-ucsc apologies for not addressing the question in the previous cycle. 
Yes, I confirmed that chalice does not strip the trailing dash. So we must account for both cases.